// @trinsic/trinsic/lib/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplateService = exports.AccountService = exports.TrustRegistryService = exports.ProviderService = exports.CredentialService = exports.WalletService = void 0;
const WalletService_1 = require("./WalletService");
Object.defineProperty(exports, "WalletService", { enumerable: true, get: function () { return WalletService_1.WalletService; } });
const ProviderService_1 = require("./ProviderService");
Object.defineProperty(exports, "ProviderService", { enumerable: true, get: function () { return ProviderService_1.ProviderService; } });
const TrustRegistryService_1 = require("./TrustRegistryService");
Object.defineProperty(exports, "TrustRegistryService", { enumerable: true, get: function () { return TrustRegistryService_1.TrustRegistryService; } });
const TemplateService_1 = require("./TemplateService");
Object.defineProperty(exports, "TemplateService", { enumerable: true, get: function () { return TemplateService_1.TemplateService; } });
const AccountService_1 = require("./AccountService");
Object.defineProperty(exports, "AccountService", { enumerable: true, get: function () { return AccountService_1.AccountService; } });
const CredentialService_1 = require("./CredentialService");
Object.defineProperty(exports, "CredentialService", { enumerable: true, get: function () { return CredentialService_1.CredentialService; } });
__exportStar(require("./proto/index"), exports);
