// @trinsic/trinsic/lib/TemplateService.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplateService = void 0;
const ServiceBase_1 = __importDefault(require("./ServiceBase"));
const proto_1 = require("./proto");
class TemplateService extends ServiceBase_1.default {
    client;
    constructor(options) {
        super(options);
        this.client = new proto_1.CredentialTemplatesClient(this.address, this.channelCredentials);
    }
    createCredentialTemplate(request) {
        return new Promise(async (resolve, reject) => {
            this.client.create(request, await this.getMetadata(request), (error, response) => {
                if (error) {
                    reject(error);
                }
                else {
                    resolve(response);
                }
            });
        });
    }
    getCredentialTemplate(request) {
        return new Promise(async (resolve, reject) => {
            this.client.get(request, await this.getMetadata(request), (error, response) => {
                if (error) {
                    reject(error);
                }
                else {
                    resolve(response);
                }
            });
        });
    }
    searchCredentialTemplate(request) {
        return new Promise(async (resolve, reject) => {
            this.client.search(request, await this.getMetadata(request), (error, response) => {
                if (error) {
                    reject(error);
                }
                else {
                    resolve(response);
                }
            });
        });
    }
    listCredentialTemplate(request) {
        return new Promise(async (resolve, reject) => {
            this.client.list(request, await this.getMetadata(request), (error, response) => {
                if (error) {
                    reject(error);
                }
                else {
                    resolve(response);
                }
            });
        });
    }
    deleteCredentialTemplate(request) {
        return new Promise(async (resolve, reject) => {
            this.client.delete(request, await this.getMetadata(request), (error, response) => {
                if (error) {
                    reject(error);
                }
                else {
                    resolve(response);
                }
            });
        });
    }
}
exports.TemplateService = TemplateService;
