// @trinsic/trinsic/lib/proto/services/common/v1/common_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var services_common_v1_common_pb = require('../../../services/common/v1/common_pb.js');
var pbmse_v1_pbmse_pb = require('../../../pbmse/v1/pbmse_pb.js');

function serialize_pbmse_v1_EncryptedMessage(arg) {
  if (!(arg instanceof pbmse_v1_pbmse_pb.EncryptedMessage)) {
    throw new Error('Expected argument of type pbmse.v1.EncryptedMessage');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_pbmse_v1_EncryptedMessage(buffer_arg) {
  return pbmse_v1_pbmse_pb.EncryptedMessage.deserializeBinary(new Uint8Array(buffer_arg));
}


var CommonService = exports.CommonService = {
  request: {
    path: '/services.common.v1.Common/Request',
    requestStream: false,
    responseStream: false,
    requestType: pbmse_v1_pbmse_pb.EncryptedMessage,
    responseType: pbmse_v1_pbmse_pb.EncryptedMessage,
    requestSerialize: serialize_pbmse_v1_EncryptedMessage,
    requestDeserialize: deserialize_pbmse_v1_EncryptedMessage,
    responseSerialize: serialize_pbmse_v1_EncryptedMessage,
    responseDeserialize: deserialize_pbmse_v1_EncryptedMessage,
  },
};

exports.CommonClient = grpc.makeGenericClientConstructor(CommonService);
