// @trinsic/trinsic/lib/proto/services/verifiable-credentials/templates/v1/templates_pb.js
// source: services/verifiable-credentials/templates/v1/templates.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.FieldType', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.GetTemplateRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.GetTemplateResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.TemplateData', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.templates.v1.TemplateField', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.displayName = 'proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.displayName = 'proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.displayName = 'proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.displayName = 'proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.displayName = 'proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.repeatedFields_, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.displayName = 'proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.displayName = 'proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.displayName = 'proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.displayName = 'proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.displayName = 'proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.TemplateField = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.TemplateField, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.TemplateField.displayName = 'proto.services.verifiablecredentials.templates.v1.TemplateField';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.GetTemplateRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.displayName = 'proto.services.verifiablecredentials.templates.v1.GetTemplateRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.GetTemplateResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.displayName = 'proto.services.verifiablecredentials.templates.v1.GetTemplateResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.displayName = 'proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.repeatedFields_, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.displayName = 'proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.templates.v1.TemplateData = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.templates.v1.TemplateData, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.templates.v1.TemplateData.displayName = 'proto.services.verifiablecredentials.templates.v1.TemplateData';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest;
  return proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    template: (f = msg.getTemplate()) && proto.services.verifiablecredentials.templates.v1.TemplateData.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse;
  return proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.services.verifiablecredentials.templates.v1.TemplateData;
      reader.readMessage(value,proto.services.verifiablecredentials.templates.v1.TemplateData.deserializeBinaryFromReader);
      msg.setTemplate(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTemplate();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.services.verifiablecredentials.templates.v1.TemplateData.serializeBinaryToWriter
    );
  }
};


/**
 * optional TemplateData template = 1;
 * @return {?proto.services.verifiablecredentials.templates.v1.TemplateData}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.prototype.getTemplate = function() {
  return /** @type{?proto.services.verifiablecredentials.templates.v1.TemplateData} */ (
    jspb.Message.getWrapperField(this, proto.services.verifiablecredentials.templates.v1.TemplateData, 1));
};


/**
 * @param {?proto.services.verifiablecredentials.templates.v1.TemplateData|undefined} value
 * @return {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse} returns this
*/
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.prototype.setTemplate = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.prototype.clearTemplate = function() {
  return this.setTemplate(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse.prototype.hasTemplate = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    query: jspb.Message.getFieldWithDefault(msg, 1, ""),
    continuationToken: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest;
  return proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setQuery(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setContinuationToken(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getQuery();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getContinuationToken();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string query = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.prototype.getQuery = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.prototype.setQuery = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string continuation_token = 2;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.prototype.getContinuationToken = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest.prototype.setContinuationToken = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    itemsJson: jspb.Message.getFieldWithDefault(msg, 1, ""),
    hasMore: jspb.Message.getBooleanFieldWithDefault(msg, 2, false),
    count: jspb.Message.getFieldWithDefault(msg, 3, 0),
    continuationToken: jspb.Message.getFieldWithDefault(msg, 4, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse;
  return proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setItemsJson(value);
      break;
    case 2:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setHasMore(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setCount(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setContinuationToken(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getItemsJson();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getHasMore();
  if (f) {
    writer.writeBool(
      2,
      f
    );
  }
  f = message.getCount();
  if (f !== 0) {
    writer.writeInt32(
      3,
      f
    );
  }
  f = message.getContinuationToken();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
};


/**
 * optional string items_json = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.getItemsJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.setItemsJson = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional bool has_more = 2;
 * @return {boolean}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.getHasMore = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 2, false));
};


/**
 * @param {boolean} value
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.setHasMore = function(value) {
  return jspb.Message.setProto3BooleanField(this, 2, value);
};


/**
 * optional int32 count = 3;
 * @return {number}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.getCount = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.setCount = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional string continuation_token = 4;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.getContinuationToken = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse.prototype.setContinuationToken = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    query: jspb.Message.getFieldWithDefault(msg, 1, ""),
    continuationToken: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest;
  return proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setQuery(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setContinuationToken(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getQuery();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getContinuationToken();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string query = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.prototype.getQuery = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.prototype.setQuery = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string continuation_token = 2;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.prototype.getContinuationToken = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest.prototype.setContinuationToken = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    templatesList: jspb.Message.toObjectList(msg.getTemplatesList(),
    proto.services.verifiablecredentials.templates.v1.TemplateData.toObject, includeInstance),
    hasMoreResults: jspb.Message.getBooleanFieldWithDefault(msg, 2, false),
    continuationToken: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse;
  return proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.services.verifiablecredentials.templates.v1.TemplateData;
      reader.readMessage(value,proto.services.verifiablecredentials.templates.v1.TemplateData.deserializeBinaryFromReader);
      msg.addTemplates(value);
      break;
    case 2:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setHasMoreResults(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setContinuationToken(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTemplatesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.services.verifiablecredentials.templates.v1.TemplateData.serializeBinaryToWriter
    );
  }
  f = message.getHasMoreResults();
  if (f) {
    writer.writeBool(
      2,
      f
    );
  }
  f = message.getContinuationToken();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * repeated TemplateData templates = 1;
 * @return {!Array<!proto.services.verifiablecredentials.templates.v1.TemplateData>}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.getTemplatesList = function() {
  return /** @type{!Array<!proto.services.verifiablecredentials.templates.v1.TemplateData>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.services.verifiablecredentials.templates.v1.TemplateData, 1));
};


/**
 * @param {!Array<!proto.services.verifiablecredentials.templates.v1.TemplateData>} value
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse} returns this
*/
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.setTemplatesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.services.verifiablecredentials.templates.v1.TemplateData=} opt_value
 * @param {number=} opt_index
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.addTemplates = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.services.verifiablecredentials.templates.v1.TemplateData, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.clearTemplatesList = function() {
  return this.setTemplatesList([]);
};


/**
 * optional bool has_more_results = 2;
 * @return {boolean}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.getHasMoreResults = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 2, false));
};


/**
 * @param {boolean} value
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.setHasMoreResults = function(value) {
  return jspb.Message.setProto3BooleanField(this, 2, value);
};


/**
 * optional string continuation_token = 3;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.getContinuationToken = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse.prototype.setContinuationToken = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest;
  return proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse;
  return proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    fieldsMap: (f = msg.getFieldsMap()) ? f.toObject(includeInstance, proto.services.verifiablecredentials.templates.v1.TemplateField.toObject) : [],
    allowAdditionalFields: jspb.Message.getBooleanFieldWithDefault(msg, 3, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest;
  return proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = msg.getFieldsMap();
      reader.readMessage(value, function(message, reader) {
        jspb.Map.deserializeBinary(message, reader, jspb.BinaryReader.prototype.readString, jspb.BinaryReader.prototype.readMessage, proto.services.verifiablecredentials.templates.v1.TemplateField.deserializeBinaryFromReader, "", new proto.services.verifiablecredentials.templates.v1.TemplateField());
         });
      break;
    case 3:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setAllowAdditionalFields(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getFieldsMap(true);
  if (f && f.getLength() > 0) {
    f.serializeBinary(2, writer, jspb.BinaryWriter.prototype.writeString, jspb.BinaryWriter.prototype.writeMessage, proto.services.verifiablecredentials.templates.v1.TemplateField.serializeBinaryToWriter);
  }
  f = message.getAllowAdditionalFields();
  if (f) {
    writer.writeBool(
      3,
      f
    );
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * map<string, TemplateField> fields = 2;
 * @param {boolean=} opt_noLazyCreate Do not create the map if
 * empty, instead returning `undefined`
 * @return {!jspb.Map<string,!proto.services.verifiablecredentials.templates.v1.TemplateField>}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.prototype.getFieldsMap = function(opt_noLazyCreate) {
  return /** @type {!jspb.Map<string,!proto.services.verifiablecredentials.templates.v1.TemplateField>} */ (
      jspb.Message.getMapField(this, 2, opt_noLazyCreate,
      proto.services.verifiablecredentials.templates.v1.TemplateField));
};


/**
 * Clears values from the map. The map will be non-null.
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.prototype.clearFieldsMap = function() {
  this.getFieldsMap().clear();
  return this;};


/**
 * optional bool allow_additional_fields = 3;
 * @return {boolean}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.prototype.getAllowAdditionalFields = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 3, false));
};


/**
 * @param {boolean} value
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest.prototype.setAllowAdditionalFields = function(value) {
  return jspb.Message.setProto3BooleanField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    data: (f = msg.getData()) && proto.services.verifiablecredentials.templates.v1.TemplateData.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse;
  return proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.services.verifiablecredentials.templates.v1.TemplateData;
      reader.readMessage(value,proto.services.verifiablecredentials.templates.v1.TemplateData.deserializeBinaryFromReader);
      msg.setData(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getData();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.services.verifiablecredentials.templates.v1.TemplateData.serializeBinaryToWriter
    );
  }
};


/**
 * optional TemplateData data = 1;
 * @return {?proto.services.verifiablecredentials.templates.v1.TemplateData}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.prototype.getData = function() {
  return /** @type{?proto.services.verifiablecredentials.templates.v1.TemplateData} */ (
    jspb.Message.getWrapperField(this, proto.services.verifiablecredentials.templates.v1.TemplateData, 1));
};


/**
 * @param {?proto.services.verifiablecredentials.templates.v1.TemplateData|undefined} value
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse} returns this
*/
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.prototype.setData = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.prototype.clearData = function() {
  return this.setData(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse.prototype.hasData = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.TemplateField.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.TemplateField} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.toObject = function(includeInstance, msg) {
  var f, obj = {
    description: jspb.Message.getFieldWithDefault(msg, 2, ""),
    optional: jspb.Message.getBooleanFieldWithDefault(msg, 3, false),
    type: jspb.Message.getFieldWithDefault(msg, 4, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateField}
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.TemplateField;
  return proto.services.verifiablecredentials.templates.v1.TemplateField.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.TemplateField} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateField}
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setDescription(value);
      break;
    case 3:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setOptional(value);
      break;
    case 4:
      var value = /** @type {!proto.services.verifiablecredentials.templates.v1.FieldType} */ (reader.readEnum());
      msg.setType(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.TemplateField.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.TemplateField} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDescription();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getOptional();
  if (f) {
    writer.writeBool(
      3,
      f
    );
  }
  f = message.getType();
  if (f !== 0.0) {
    writer.writeEnum(
      4,
      f
    );
  }
};


/**
 * optional string description = 2;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.prototype.getDescription = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateField} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.prototype.setDescription = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional bool optional = 3;
 * @return {boolean}
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.prototype.getOptional = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 3, false));
};


/**
 * @param {boolean} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateField} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.prototype.setOptional = function(value) {
  return jspb.Message.setProto3BooleanField(this, 3, value);
};


/**
 * optional FieldType type = 4;
 * @return {!proto.services.verifiablecredentials.templates.v1.FieldType}
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.prototype.getType = function() {
  return /** @type {!proto.services.verifiablecredentials.templates.v1.FieldType} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {!proto.services.verifiablecredentials.templates.v1.FieldType} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateField} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateField.prototype.setType = function(value) {
  return jspb.Message.setProto3EnumField(this, 4, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.GetTemplateRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetTemplateRequest}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.GetTemplateRequest;
  return proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.GetTemplateRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetTemplateRequest}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.GetTemplateRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.GetTemplateRequest} returns this
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateRequest.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.GetTemplateResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    data: (f = msg.getData()) && proto.services.verifiablecredentials.templates.v1.TemplateData.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetTemplateResponse}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.GetTemplateResponse;
  return proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.GetTemplateResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetTemplateResponse}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.services.verifiablecredentials.templates.v1.TemplateData;
      reader.readMessage(value,proto.services.verifiablecredentials.templates.v1.TemplateData.deserializeBinaryFromReader);
      msg.setData(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.GetTemplateResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getData();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.services.verifiablecredentials.templates.v1.TemplateData.serializeBinaryToWriter
    );
  }
};


/**
 * optional TemplateData data = 1;
 * @return {?proto.services.verifiablecredentials.templates.v1.TemplateData}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.prototype.getData = function() {
  return /** @type{?proto.services.verifiablecredentials.templates.v1.TemplateData} */ (
    jspb.Message.getWrapperField(this, proto.services.verifiablecredentials.templates.v1.TemplateData, 1));
};


/**
 * @param {?proto.services.verifiablecredentials.templates.v1.TemplateData|undefined} value
 * @return {!proto.services.verifiablecredentials.templates.v1.GetTemplateResponse} returns this
*/
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.prototype.setData = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.services.verifiablecredentials.templates.v1.GetTemplateResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.prototype.clearData = function() {
  return this.setData(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.services.verifiablecredentials.templates.v1.GetTemplateResponse.prototype.hasData = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest;
  return proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    templatesList: jspb.Message.toObjectList(msg.getTemplatesList(),
    proto.services.verifiablecredentials.templates.v1.TemplateData.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse;
  return proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.services.verifiablecredentials.templates.v1.TemplateData;
      reader.readMessage(value,proto.services.verifiablecredentials.templates.v1.TemplateData.deserializeBinaryFromReader);
      msg.addTemplates(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTemplatesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.services.verifiablecredentials.templates.v1.TemplateData.serializeBinaryToWriter
    );
  }
};


/**
 * repeated TemplateData templates = 1;
 * @return {!Array<!proto.services.verifiablecredentials.templates.v1.TemplateData>}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.prototype.getTemplatesList = function() {
  return /** @type{!Array<!proto.services.verifiablecredentials.templates.v1.TemplateData>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.services.verifiablecredentials.templates.v1.TemplateData, 1));
};


/**
 * @param {!Array<!proto.services.verifiablecredentials.templates.v1.TemplateData>} value
 * @return {!proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse} returns this
*/
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.prototype.setTemplatesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.services.verifiablecredentials.templates.v1.TemplateData=} opt_value
 * @param {number=} opt_index
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData}
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.prototype.addTemplates = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.services.verifiablecredentials.templates.v1.TemplateData, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse} returns this
 */
proto.services.verifiablecredentials.templates.v1.ListTemplatesResponse.prototype.clearTemplatesList = function() {
  return this.setTemplatesList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.templates.v1.TemplateData.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.templates.v1.TemplateData} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    name: jspb.Message.getFieldWithDefault(msg, 2, ""),
    version: jspb.Message.getFieldWithDefault(msg, 3, 0),
    fieldsMap: (f = msg.getFieldsMap()) ? f.toObject(includeInstance, proto.services.verifiablecredentials.templates.v1.TemplateField.toObject) : [],
    allowAdditionalFields: jspb.Message.getBooleanFieldWithDefault(msg, 5, false),
    schemaUri: jspb.Message.getFieldWithDefault(msg, 6, ""),
    contextUri: jspb.Message.getFieldWithDefault(msg, 7, ""),
    ecosystemId: jspb.Message.getFieldWithDefault(msg, 8, ""),
    type: jspb.Message.getFieldWithDefault(msg, 9, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.templates.v1.TemplateData;
  return proto.services.verifiablecredentials.templates.v1.TemplateData.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.templates.v1.TemplateData} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setVersion(value);
      break;
    case 4:
      var value = msg.getFieldsMap();
      reader.readMessage(value, function(message, reader) {
        jspb.Map.deserializeBinary(message, reader, jspb.BinaryReader.prototype.readString, jspb.BinaryReader.prototype.readMessage, proto.services.verifiablecredentials.templates.v1.TemplateField.deserializeBinaryFromReader, "", new proto.services.verifiablecredentials.templates.v1.TemplateField());
         });
      break;
    case 5:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setAllowAdditionalFields(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setSchemaUri(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setContextUri(value);
      break;
    case 8:
      var value = /** @type {string} */ (reader.readString());
      msg.setEcosystemId(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readString());
      msg.setType(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.templates.v1.TemplateData.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.templates.v1.TemplateData} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getVersion();
  if (f !== 0) {
    writer.writeInt32(
      3,
      f
    );
  }
  f = message.getFieldsMap(true);
  if (f && f.getLength() > 0) {
    f.serializeBinary(4, writer, jspb.BinaryWriter.prototype.writeString, jspb.BinaryWriter.prototype.writeMessage, proto.services.verifiablecredentials.templates.v1.TemplateField.serializeBinaryToWriter);
  }
  f = message.getAllowAdditionalFields();
  if (f) {
    writer.writeBool(
      5,
      f
    );
  }
  f = message.getSchemaUri();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getContextUri();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
  f = message.getEcosystemId();
  if (f.length > 0) {
    writer.writeString(
      8,
      f
    );
  }
  f = message.getType();
  if (f.length > 0) {
    writer.writeString(
      9,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string name = 2;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional int32 version = 3;
 * @return {number}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getVersion = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.setVersion = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * map<string, TemplateField> fields = 4;
 * @param {boolean=} opt_noLazyCreate Do not create the map if
 * empty, instead returning `undefined`
 * @return {!jspb.Map<string,!proto.services.verifiablecredentials.templates.v1.TemplateField>}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getFieldsMap = function(opt_noLazyCreate) {
  return /** @type {!jspb.Map<string,!proto.services.verifiablecredentials.templates.v1.TemplateField>} */ (
      jspb.Message.getMapField(this, 4, opt_noLazyCreate,
      proto.services.verifiablecredentials.templates.v1.TemplateField));
};


/**
 * Clears values from the map. The map will be non-null.
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.clearFieldsMap = function() {
  this.getFieldsMap().clear();
  return this;};


/**
 * optional bool allow_additional_fields = 5;
 * @return {boolean}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getAllowAdditionalFields = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 5, false));
};


/**
 * @param {boolean} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.setAllowAdditionalFields = function(value) {
  return jspb.Message.setProto3BooleanField(this, 5, value);
};


/**
 * optional string schema_uri = 6;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getSchemaUri = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.setSchemaUri = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional string context_uri = 7;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getContextUri = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.setContextUri = function(value) {
  return jspb.Message.setProto3StringField(this, 7, value);
};


/**
 * optional string ecosystem_id = 8;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getEcosystemId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.setEcosystemId = function(value) {
  return jspb.Message.setProto3StringField(this, 8, value);
};


/**
 * optional string type = 9;
 * @return {string}
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.getType = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 9, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.templates.v1.TemplateData} returns this
 */
proto.services.verifiablecredentials.templates.v1.TemplateData.prototype.setType = function(value) {
  return jspb.Message.setProto3StringField(this, 9, value);
};


/**
 * @enum {number}
 */
proto.services.verifiablecredentials.templates.v1.FieldType = {
  STRING: 0,
  NUMBER: 1,
  BOOL: 2,
  DATETIME: 4
};

goog.object.extend(exports, proto.services.verifiablecredentials.templates.v1);
