// @trinsic/trinsic/lib/proto/services/verifiable-credentials/templates/v1/templates_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var services_verifiable$credentials_templates_v1_templates_pb = require('../../../../services/verifiable-credentials/templates/v1/templates_pb.js');

function serialize_services_verifiablecredentials_templates_v1_CreateCredentialTemplateRequest(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.CreateCredentialTemplateRequest)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.CreateCredentialTemplateRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_CreateCredentialTemplateRequest(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.CreateCredentialTemplateRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_CreateCredentialTemplateResponse(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.CreateCredentialTemplateResponse)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.CreateCredentialTemplateResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_CreateCredentialTemplateResponse(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.CreateCredentialTemplateResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_DeleteCredentialTemplateRequest(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.DeleteCredentialTemplateRequest)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.DeleteCredentialTemplateRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_DeleteCredentialTemplateRequest(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.DeleteCredentialTemplateRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_DeleteCredentialTemplateResponse(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.DeleteCredentialTemplateResponse)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.DeleteCredentialTemplateResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_DeleteCredentialTemplateResponse(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.DeleteCredentialTemplateResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_GetCredentialTemplateRequest(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.GetCredentialTemplateRequest)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.GetCredentialTemplateRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_GetCredentialTemplateRequest(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.GetCredentialTemplateRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_GetCredentialTemplateResponse(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.GetCredentialTemplateResponse)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.GetCredentialTemplateResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_GetCredentialTemplateResponse(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.GetCredentialTemplateResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_ListCredentialTemplatesRequest(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.ListCredentialTemplatesRequest)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.ListCredentialTemplatesRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_ListCredentialTemplatesRequest(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.ListCredentialTemplatesRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_ListCredentialTemplatesResponse(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.ListCredentialTemplatesResponse)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.ListCredentialTemplatesResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_ListCredentialTemplatesResponse(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.ListCredentialTemplatesResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_SearchCredentialTemplatesRequest(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.SearchCredentialTemplatesRequest)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.SearchCredentialTemplatesRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_SearchCredentialTemplatesRequest(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.SearchCredentialTemplatesRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_services_verifiablecredentials_templates_v1_SearchCredentialTemplatesResponse(arg) {
  if (!(arg instanceof services_verifiable$credentials_templates_v1_templates_pb.SearchCredentialTemplatesResponse)) {
    throw new Error('Expected argument of type services.verifiablecredentials.templates.v1.SearchCredentialTemplatesResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_services_verifiablecredentials_templates_v1_SearchCredentialTemplatesResponse(buffer_arg) {
  return services_verifiable$credentials_templates_v1_templates_pb.SearchCredentialTemplatesResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var CredentialTemplatesService = exports.CredentialTemplatesService = {
  create: {
    path: '/services.verifiablecredentials.templates.v1.CredentialTemplates/Create',
    requestStream: false,
    responseStream: false,
    requestType: services_verifiable$credentials_templates_v1_templates_pb.CreateCredentialTemplateRequest,
    responseType: services_verifiable$credentials_templates_v1_templates_pb.CreateCredentialTemplateResponse,
    requestSerialize: serialize_services_verifiablecredentials_templates_v1_CreateCredentialTemplateRequest,
    requestDeserialize: deserialize_services_verifiablecredentials_templates_v1_CreateCredentialTemplateRequest,
    responseSerialize: serialize_services_verifiablecredentials_templates_v1_CreateCredentialTemplateResponse,
    responseDeserialize: deserialize_services_verifiablecredentials_templates_v1_CreateCredentialTemplateResponse,
  },
  get: {
    path: '/services.verifiablecredentials.templates.v1.CredentialTemplates/Get',
    requestStream: false,
    responseStream: false,
    requestType: services_verifiable$credentials_templates_v1_templates_pb.GetCredentialTemplateRequest,
    responseType: services_verifiable$credentials_templates_v1_templates_pb.GetCredentialTemplateResponse,
    requestSerialize: serialize_services_verifiablecredentials_templates_v1_GetCredentialTemplateRequest,
    requestDeserialize: deserialize_services_verifiablecredentials_templates_v1_GetCredentialTemplateRequest,
    responseSerialize: serialize_services_verifiablecredentials_templates_v1_GetCredentialTemplateResponse,
    responseDeserialize: deserialize_services_verifiablecredentials_templates_v1_GetCredentialTemplateResponse,
  },
  list: {
    path: '/services.verifiablecredentials.templates.v1.CredentialTemplates/List',
    requestStream: false,
    responseStream: false,
    requestType: services_verifiable$credentials_templates_v1_templates_pb.ListCredentialTemplatesRequest,
    responseType: services_verifiable$credentials_templates_v1_templates_pb.ListCredentialTemplatesResponse,
    requestSerialize: serialize_services_verifiablecredentials_templates_v1_ListCredentialTemplatesRequest,
    requestDeserialize: deserialize_services_verifiablecredentials_templates_v1_ListCredentialTemplatesRequest,
    responseSerialize: serialize_services_verifiablecredentials_templates_v1_ListCredentialTemplatesResponse,
    responseDeserialize: deserialize_services_verifiablecredentials_templates_v1_ListCredentialTemplatesResponse,
  },
  search: {
    path: '/services.verifiablecredentials.templates.v1.CredentialTemplates/Search',
    requestStream: false,
    responseStream: false,
    requestType: services_verifiable$credentials_templates_v1_templates_pb.SearchCredentialTemplatesRequest,
    responseType: services_verifiable$credentials_templates_v1_templates_pb.SearchCredentialTemplatesResponse,
    requestSerialize: serialize_services_verifiablecredentials_templates_v1_SearchCredentialTemplatesRequest,
    requestDeserialize: deserialize_services_verifiablecredentials_templates_v1_SearchCredentialTemplatesRequest,
    responseSerialize: serialize_services_verifiablecredentials_templates_v1_SearchCredentialTemplatesResponse,
    responseDeserialize: deserialize_services_verifiablecredentials_templates_v1_SearchCredentialTemplatesResponse,
  },
  delete: {
    path: '/services.verifiablecredentials.templates.v1.CredentialTemplates/Delete',
    requestStream: false,
    responseStream: false,
    requestType: services_verifiable$credentials_templates_v1_templates_pb.DeleteCredentialTemplateRequest,
    responseType: services_verifiable$credentials_templates_v1_templates_pb.DeleteCredentialTemplateResponse,
    requestSerialize: serialize_services_verifiablecredentials_templates_v1_DeleteCredentialTemplateRequest,
    requestDeserialize: deserialize_services_verifiablecredentials_templates_v1_DeleteCredentialTemplateRequest,
    responseSerialize: serialize_services_verifiablecredentials_templates_v1_DeleteCredentialTemplateResponse,
    responseDeserialize: deserialize_services_verifiablecredentials_templates_v1_DeleteCredentialTemplateResponse,
  },
};

exports.CredentialTemplatesClient = grpc.makeGenericClientConstructor(CredentialTemplatesService);
