// @trinsic/okapi-proto/lib/proto/okapi/transport/v1/transport_pb.js
// source: okapi/transport/v1/transport.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var okapi_keys_v1_keys_pb = require('../../../okapi/keys/v1/keys_pb.js');
goog.object.extend(proto, okapi_keys_v1_keys_pb);
var pbmse_v1_pbmse_pb = require('../../../pbmse/v1/pbmse_pb.js');
goog.object.extend(proto, pbmse_v1_pbmse_pb);
goog.exportSymbol('proto.okapi.transport.v1.CoreMessage', null, global);
goog.exportSymbol('proto.okapi.transport.v1.PackRequest', null, global);
goog.exportSymbol('proto.okapi.transport.v1.PackResponse', null, global);
goog.exportSymbol('proto.okapi.transport.v1.SignRequest', null, global);
goog.exportSymbol('proto.okapi.transport.v1.SignResponse', null, global);
goog.exportSymbol('proto.okapi.transport.v1.UnpackRequest', null, global);
goog.exportSymbol('proto.okapi.transport.v1.UnpackResponse', null, global);
goog.exportSymbol('proto.okapi.transport.v1.VerifyRequest', null, global);
goog.exportSymbol('proto.okapi.transport.v1.VerifyResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.SignRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.okapi.transport.v1.SignRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.SignRequest.displayName = 'proto.okapi.transport.v1.SignRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.SignResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.okapi.transport.v1.SignResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.SignResponse.displayName = 'proto.okapi.transport.v1.SignResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.VerifyRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.okapi.transport.v1.VerifyRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.VerifyRequest.displayName = 'proto.okapi.transport.v1.VerifyRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.VerifyResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.okapi.transport.v1.VerifyResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.VerifyResponse.displayName = 'proto.okapi.transport.v1.VerifyResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.PackRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.okapi.transport.v1.PackRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.PackRequest.displayName = 'proto.okapi.transport.v1.PackRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.PackResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.okapi.transport.v1.PackResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.PackResponse.displayName = 'proto.okapi.transport.v1.PackResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.UnpackRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.okapi.transport.v1.UnpackRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.UnpackRequest.displayName = 'proto.okapi.transport.v1.UnpackRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.UnpackResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.okapi.transport.v1.UnpackResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.UnpackResponse.displayName = 'proto.okapi.transport.v1.UnpackResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.okapi.transport.v1.CoreMessage = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.okapi.transport.v1.CoreMessage.repeatedFields_, null);
};
goog.inherits(proto.okapi.transport.v1.CoreMessage, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.okapi.transport.v1.CoreMessage.displayName = 'proto.okapi.transport.v1.CoreMessage';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.SignRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.SignRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.SignRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.SignRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    payload: msg.getPayload_asB64(),
    key: (f = msg.getKey()) && okapi_keys_v1_keys_pb.JsonWebKey.toObject(includeInstance, f),
    appendTo: (f = msg.getAppendTo()) && pbmse_v1_pbmse_pb.SignedMessage.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.SignRequest}
 */
proto.okapi.transport.v1.SignRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.SignRequest;
  return proto.okapi.transport.v1.SignRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.SignRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.SignRequest}
 */
proto.okapi.transport.v1.SignRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setPayload(value);
      break;
    case 2:
      var value = new okapi_keys_v1_keys_pb.JsonWebKey;
      reader.readMessage(value,okapi_keys_v1_keys_pb.JsonWebKey.deserializeBinaryFromReader);
      msg.setKey(value);
      break;
    case 3:
      var value = new pbmse_v1_pbmse_pb.SignedMessage;
      reader.readMessage(value,pbmse_v1_pbmse_pb.SignedMessage.deserializeBinaryFromReader);
      msg.setAppendTo(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.SignRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.SignRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.SignRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.SignRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getPayload_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      1,
      f
    );
  }
  f = message.getKey();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      okapi_keys_v1_keys_pb.JsonWebKey.serializeBinaryToWriter
    );
  }
  f = message.getAppendTo();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      pbmse_v1_pbmse_pb.SignedMessage.serializeBinaryToWriter
    );
  }
};


/**
 * optional bytes payload = 1;
 * @return {!(string|Uint8Array)}
 */
proto.okapi.transport.v1.SignRequest.prototype.getPayload = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * optional bytes payload = 1;
 * This is a type-conversion wrapper around `getPayload()`
 * @return {string}
 */
proto.okapi.transport.v1.SignRequest.prototype.getPayload_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getPayload()));
};


/**
 * optional bytes payload = 1;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getPayload()`
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.SignRequest.prototype.getPayload_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getPayload()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.okapi.transport.v1.SignRequest} returns this
 */
proto.okapi.transport.v1.SignRequest.prototype.setPayload = function(value) {
  return jspb.Message.setProto3BytesField(this, 1, value);
};


/**
 * optional okapi.keys.v1.JsonWebKey key = 2;
 * @return {?proto.okapi.keys.v1.JsonWebKey}
 */
proto.okapi.transport.v1.SignRequest.prototype.getKey = function() {
  return /** @type{?proto.okapi.keys.v1.JsonWebKey} */ (
    jspb.Message.getWrapperField(this, okapi_keys_v1_keys_pb.JsonWebKey, 2));
};


/**
 * @param {?proto.okapi.keys.v1.JsonWebKey|undefined} value
 * @return {!proto.okapi.transport.v1.SignRequest} returns this
*/
proto.okapi.transport.v1.SignRequest.prototype.setKey = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.SignRequest} returns this
 */
proto.okapi.transport.v1.SignRequest.prototype.clearKey = function() {
  return this.setKey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.SignRequest.prototype.hasKey = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional pbmse.v1.SignedMessage append_to = 3;
 * @return {?proto.pbmse.v1.SignedMessage}
 */
proto.okapi.transport.v1.SignRequest.prototype.getAppendTo = function() {
  return /** @type{?proto.pbmse.v1.SignedMessage} */ (
    jspb.Message.getWrapperField(this, pbmse_v1_pbmse_pb.SignedMessage, 3));
};


/**
 * @param {?proto.pbmse.v1.SignedMessage|undefined} value
 * @return {!proto.okapi.transport.v1.SignRequest} returns this
*/
proto.okapi.transport.v1.SignRequest.prototype.setAppendTo = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.SignRequest} returns this
 */
proto.okapi.transport.v1.SignRequest.prototype.clearAppendTo = function() {
  return this.setAppendTo(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.SignRequest.prototype.hasAppendTo = function() {
  return jspb.Message.getField(this, 3) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.SignResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.SignResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.SignResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.SignResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    message: (f = msg.getMessage()) && pbmse_v1_pbmse_pb.SignedMessage.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.SignResponse}
 */
proto.okapi.transport.v1.SignResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.SignResponse;
  return proto.okapi.transport.v1.SignResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.SignResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.SignResponse}
 */
proto.okapi.transport.v1.SignResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new pbmse_v1_pbmse_pb.SignedMessage;
      reader.readMessage(value,pbmse_v1_pbmse_pb.SignedMessage.deserializeBinaryFromReader);
      msg.setMessage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.SignResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.SignResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.SignResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.SignResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMessage();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      pbmse_v1_pbmse_pb.SignedMessage.serializeBinaryToWriter
    );
  }
};


/**
 * optional pbmse.v1.SignedMessage message = 1;
 * @return {?proto.pbmse.v1.SignedMessage}
 */
proto.okapi.transport.v1.SignResponse.prototype.getMessage = function() {
  return /** @type{?proto.pbmse.v1.SignedMessage} */ (
    jspb.Message.getWrapperField(this, pbmse_v1_pbmse_pb.SignedMessage, 1));
};


/**
 * @param {?proto.pbmse.v1.SignedMessage|undefined} value
 * @return {!proto.okapi.transport.v1.SignResponse} returns this
*/
proto.okapi.transport.v1.SignResponse.prototype.setMessage = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.SignResponse} returns this
 */
proto.okapi.transport.v1.SignResponse.prototype.clearMessage = function() {
  return this.setMessage(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.SignResponse.prototype.hasMessage = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.VerifyRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.VerifyRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.VerifyRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.VerifyRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    message: (f = msg.getMessage()) && pbmse_v1_pbmse_pb.SignedMessage.toObject(includeInstance, f),
    key: (f = msg.getKey()) && okapi_keys_v1_keys_pb.JsonWebKey.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.VerifyRequest}
 */
proto.okapi.transport.v1.VerifyRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.VerifyRequest;
  return proto.okapi.transport.v1.VerifyRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.VerifyRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.VerifyRequest}
 */
proto.okapi.transport.v1.VerifyRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new pbmse_v1_pbmse_pb.SignedMessage;
      reader.readMessage(value,pbmse_v1_pbmse_pb.SignedMessage.deserializeBinaryFromReader);
      msg.setMessage(value);
      break;
    case 2:
      var value = new okapi_keys_v1_keys_pb.JsonWebKey;
      reader.readMessage(value,okapi_keys_v1_keys_pb.JsonWebKey.deserializeBinaryFromReader);
      msg.setKey(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.VerifyRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.VerifyRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.VerifyRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.VerifyRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMessage();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      pbmse_v1_pbmse_pb.SignedMessage.serializeBinaryToWriter
    );
  }
  f = message.getKey();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      okapi_keys_v1_keys_pb.JsonWebKey.serializeBinaryToWriter
    );
  }
};


/**
 * optional pbmse.v1.SignedMessage message = 1;
 * @return {?proto.pbmse.v1.SignedMessage}
 */
proto.okapi.transport.v1.VerifyRequest.prototype.getMessage = function() {
  return /** @type{?proto.pbmse.v1.SignedMessage} */ (
    jspb.Message.getWrapperField(this, pbmse_v1_pbmse_pb.SignedMessage, 1));
};


/**
 * @param {?proto.pbmse.v1.SignedMessage|undefined} value
 * @return {!proto.okapi.transport.v1.VerifyRequest} returns this
*/
proto.okapi.transport.v1.VerifyRequest.prototype.setMessage = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.VerifyRequest} returns this
 */
proto.okapi.transport.v1.VerifyRequest.prototype.clearMessage = function() {
  return this.setMessage(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.VerifyRequest.prototype.hasMessage = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional okapi.keys.v1.JsonWebKey key = 2;
 * @return {?proto.okapi.keys.v1.JsonWebKey}
 */
proto.okapi.transport.v1.VerifyRequest.prototype.getKey = function() {
  return /** @type{?proto.okapi.keys.v1.JsonWebKey} */ (
    jspb.Message.getWrapperField(this, okapi_keys_v1_keys_pb.JsonWebKey, 2));
};


/**
 * @param {?proto.okapi.keys.v1.JsonWebKey|undefined} value
 * @return {!proto.okapi.transport.v1.VerifyRequest} returns this
*/
proto.okapi.transport.v1.VerifyRequest.prototype.setKey = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.VerifyRequest} returns this
 */
proto.okapi.transport.v1.VerifyRequest.prototype.clearKey = function() {
  return this.setKey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.VerifyRequest.prototype.hasKey = function() {
  return jspb.Message.getField(this, 2) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.VerifyResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.VerifyResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.VerifyResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.VerifyResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    isValid: jspb.Message.getBooleanFieldWithDefault(msg, 1, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.VerifyResponse}
 */
proto.okapi.transport.v1.VerifyResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.VerifyResponse;
  return proto.okapi.transport.v1.VerifyResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.VerifyResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.VerifyResponse}
 */
proto.okapi.transport.v1.VerifyResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setIsValid(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.VerifyResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.VerifyResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.VerifyResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.VerifyResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getIsValid();
  if (f) {
    writer.writeBool(
      1,
      f
    );
  }
};


/**
 * optional bool is_valid = 1;
 * @return {boolean}
 */
proto.okapi.transport.v1.VerifyResponse.prototype.getIsValid = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 1, false));
};


/**
 * @param {boolean} value
 * @return {!proto.okapi.transport.v1.VerifyResponse} returns this
 */
proto.okapi.transport.v1.VerifyResponse.prototype.setIsValid = function(value) {
  return jspb.Message.setProto3BooleanField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.PackRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.PackRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.PackRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.PackRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    senderKey: (f = msg.getSenderKey()) && okapi_keys_v1_keys_pb.JsonWebKey.toObject(includeInstance, f),
    receiverKey: (f = msg.getReceiverKey()) && okapi_keys_v1_keys_pb.JsonWebKey.toObject(includeInstance, f),
    associatedData: msg.getAssociatedData_asB64(),
    plaintext: msg.getPlaintext_asB64(),
    mode: jspb.Message.getFieldWithDefault(msg, 5, 0),
    algorithm: jspb.Message.getFieldWithDefault(msg, 6, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.PackRequest}
 */
proto.okapi.transport.v1.PackRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.PackRequest;
  return proto.okapi.transport.v1.PackRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.PackRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.PackRequest}
 */
proto.okapi.transport.v1.PackRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new okapi_keys_v1_keys_pb.JsonWebKey;
      reader.readMessage(value,okapi_keys_v1_keys_pb.JsonWebKey.deserializeBinaryFromReader);
      msg.setSenderKey(value);
      break;
    case 2:
      var value = new okapi_keys_v1_keys_pb.JsonWebKey;
      reader.readMessage(value,okapi_keys_v1_keys_pb.JsonWebKey.deserializeBinaryFromReader);
      msg.setReceiverKey(value);
      break;
    case 3:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setAssociatedData(value);
      break;
    case 4:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setPlaintext(value);
      break;
    case 5:
      var value = /** @type {!proto.pbmse.v1.EncryptionMode} */ (reader.readEnum());
      msg.setMode(value);
      break;
    case 6:
      var value = /** @type {!proto.pbmse.v1.EncryptionAlgorithm} */ (reader.readEnum());
      msg.setAlgorithm(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.PackRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.PackRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.PackRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.PackRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSenderKey();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      okapi_keys_v1_keys_pb.JsonWebKey.serializeBinaryToWriter
    );
  }
  f = message.getReceiverKey();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      okapi_keys_v1_keys_pb.JsonWebKey.serializeBinaryToWriter
    );
  }
  f = message.getAssociatedData_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      3,
      f
    );
  }
  f = message.getPlaintext_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      4,
      f
    );
  }
  f = message.getMode();
  if (f !== 0.0) {
    writer.writeEnum(
      5,
      f
    );
  }
  f = message.getAlgorithm();
  if (f !== 0.0) {
    writer.writeEnum(
      6,
      f
    );
  }
};


/**
 * optional okapi.keys.v1.JsonWebKey sender_key = 1;
 * @return {?proto.okapi.keys.v1.JsonWebKey}
 */
proto.okapi.transport.v1.PackRequest.prototype.getSenderKey = function() {
  return /** @type{?proto.okapi.keys.v1.JsonWebKey} */ (
    jspb.Message.getWrapperField(this, okapi_keys_v1_keys_pb.JsonWebKey, 1));
};


/**
 * @param {?proto.okapi.keys.v1.JsonWebKey|undefined} value
 * @return {!proto.okapi.transport.v1.PackRequest} returns this
*/
proto.okapi.transport.v1.PackRequest.prototype.setSenderKey = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.PackRequest} returns this
 */
proto.okapi.transport.v1.PackRequest.prototype.clearSenderKey = function() {
  return this.setSenderKey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.PackRequest.prototype.hasSenderKey = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional okapi.keys.v1.JsonWebKey receiver_key = 2;
 * @return {?proto.okapi.keys.v1.JsonWebKey}
 */
proto.okapi.transport.v1.PackRequest.prototype.getReceiverKey = function() {
  return /** @type{?proto.okapi.keys.v1.JsonWebKey} */ (
    jspb.Message.getWrapperField(this, okapi_keys_v1_keys_pb.JsonWebKey, 2));
};


/**
 * @param {?proto.okapi.keys.v1.JsonWebKey|undefined} value
 * @return {!proto.okapi.transport.v1.PackRequest} returns this
*/
proto.okapi.transport.v1.PackRequest.prototype.setReceiverKey = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.PackRequest} returns this
 */
proto.okapi.transport.v1.PackRequest.prototype.clearReceiverKey = function() {
  return this.setReceiverKey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.PackRequest.prototype.hasReceiverKey = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional bytes associated_data = 3;
 * @return {!(string|Uint8Array)}
 */
proto.okapi.transport.v1.PackRequest.prototype.getAssociatedData = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * optional bytes associated_data = 3;
 * This is a type-conversion wrapper around `getAssociatedData()`
 * @return {string}
 */
proto.okapi.transport.v1.PackRequest.prototype.getAssociatedData_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getAssociatedData()));
};


/**
 * optional bytes associated_data = 3;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getAssociatedData()`
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.PackRequest.prototype.getAssociatedData_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getAssociatedData()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.okapi.transport.v1.PackRequest} returns this
 */
proto.okapi.transport.v1.PackRequest.prototype.setAssociatedData = function(value) {
  return jspb.Message.setProto3BytesField(this, 3, value);
};


/**
 * optional bytes plaintext = 4;
 * @return {!(string|Uint8Array)}
 */
proto.okapi.transport.v1.PackRequest.prototype.getPlaintext = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * optional bytes plaintext = 4;
 * This is a type-conversion wrapper around `getPlaintext()`
 * @return {string}
 */
proto.okapi.transport.v1.PackRequest.prototype.getPlaintext_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getPlaintext()));
};


/**
 * optional bytes plaintext = 4;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getPlaintext()`
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.PackRequest.prototype.getPlaintext_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getPlaintext()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.okapi.transport.v1.PackRequest} returns this
 */
proto.okapi.transport.v1.PackRequest.prototype.setPlaintext = function(value) {
  return jspb.Message.setProto3BytesField(this, 4, value);
};


/**
 * optional pbmse.v1.EncryptionMode mode = 5;
 * @return {!proto.pbmse.v1.EncryptionMode}
 */
proto.okapi.transport.v1.PackRequest.prototype.getMode = function() {
  return /** @type {!proto.pbmse.v1.EncryptionMode} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {!proto.pbmse.v1.EncryptionMode} value
 * @return {!proto.okapi.transport.v1.PackRequest} returns this
 */
proto.okapi.transport.v1.PackRequest.prototype.setMode = function(value) {
  return jspb.Message.setProto3EnumField(this, 5, value);
};


/**
 * optional pbmse.v1.EncryptionAlgorithm algorithm = 6;
 * @return {!proto.pbmse.v1.EncryptionAlgorithm}
 */
proto.okapi.transport.v1.PackRequest.prototype.getAlgorithm = function() {
  return /** @type {!proto.pbmse.v1.EncryptionAlgorithm} */ (jspb.Message.getFieldWithDefault(this, 6, 0));
};


/**
 * @param {!proto.pbmse.v1.EncryptionAlgorithm} value
 * @return {!proto.okapi.transport.v1.PackRequest} returns this
 */
proto.okapi.transport.v1.PackRequest.prototype.setAlgorithm = function(value) {
  return jspb.Message.setProto3EnumField(this, 6, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.PackResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.PackResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.PackResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.PackResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    message: (f = msg.getMessage()) && pbmse_v1_pbmse_pb.EncryptedMessage.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.PackResponse}
 */
proto.okapi.transport.v1.PackResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.PackResponse;
  return proto.okapi.transport.v1.PackResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.PackResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.PackResponse}
 */
proto.okapi.transport.v1.PackResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new pbmse_v1_pbmse_pb.EncryptedMessage;
      reader.readMessage(value,pbmse_v1_pbmse_pb.EncryptedMessage.deserializeBinaryFromReader);
      msg.setMessage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.PackResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.PackResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.PackResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.PackResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getMessage();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      pbmse_v1_pbmse_pb.EncryptedMessage.serializeBinaryToWriter
    );
  }
};


/**
 * optional pbmse.v1.EncryptedMessage message = 1;
 * @return {?proto.pbmse.v1.EncryptedMessage}
 */
proto.okapi.transport.v1.PackResponse.prototype.getMessage = function() {
  return /** @type{?proto.pbmse.v1.EncryptedMessage} */ (
    jspb.Message.getWrapperField(this, pbmse_v1_pbmse_pb.EncryptedMessage, 1));
};


/**
 * @param {?proto.pbmse.v1.EncryptedMessage|undefined} value
 * @return {!proto.okapi.transport.v1.PackResponse} returns this
*/
proto.okapi.transport.v1.PackResponse.prototype.setMessage = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.PackResponse} returns this
 */
proto.okapi.transport.v1.PackResponse.prototype.clearMessage = function() {
  return this.setMessage(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.PackResponse.prototype.hasMessage = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.UnpackRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.UnpackRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.UnpackRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.UnpackRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    senderKey: (f = msg.getSenderKey()) && okapi_keys_v1_keys_pb.JsonWebKey.toObject(includeInstance, f),
    receiverKey: (f = msg.getReceiverKey()) && okapi_keys_v1_keys_pb.JsonWebKey.toObject(includeInstance, f),
    message: (f = msg.getMessage()) && pbmse_v1_pbmse_pb.EncryptedMessage.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.UnpackRequest}
 */
proto.okapi.transport.v1.UnpackRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.UnpackRequest;
  return proto.okapi.transport.v1.UnpackRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.UnpackRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.UnpackRequest}
 */
proto.okapi.transport.v1.UnpackRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new okapi_keys_v1_keys_pb.JsonWebKey;
      reader.readMessage(value,okapi_keys_v1_keys_pb.JsonWebKey.deserializeBinaryFromReader);
      msg.setSenderKey(value);
      break;
    case 2:
      var value = new okapi_keys_v1_keys_pb.JsonWebKey;
      reader.readMessage(value,okapi_keys_v1_keys_pb.JsonWebKey.deserializeBinaryFromReader);
      msg.setReceiverKey(value);
      break;
    case 3:
      var value = new pbmse_v1_pbmse_pb.EncryptedMessage;
      reader.readMessage(value,pbmse_v1_pbmse_pb.EncryptedMessage.deserializeBinaryFromReader);
      msg.setMessage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.UnpackRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.UnpackRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.UnpackRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.UnpackRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSenderKey();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      okapi_keys_v1_keys_pb.JsonWebKey.serializeBinaryToWriter
    );
  }
  f = message.getReceiverKey();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      okapi_keys_v1_keys_pb.JsonWebKey.serializeBinaryToWriter
    );
  }
  f = message.getMessage();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      pbmse_v1_pbmse_pb.EncryptedMessage.serializeBinaryToWriter
    );
  }
};


/**
 * optional okapi.keys.v1.JsonWebKey sender_key = 1;
 * @return {?proto.okapi.keys.v1.JsonWebKey}
 */
proto.okapi.transport.v1.UnpackRequest.prototype.getSenderKey = function() {
  return /** @type{?proto.okapi.keys.v1.JsonWebKey} */ (
    jspb.Message.getWrapperField(this, okapi_keys_v1_keys_pb.JsonWebKey, 1));
};


/**
 * @param {?proto.okapi.keys.v1.JsonWebKey|undefined} value
 * @return {!proto.okapi.transport.v1.UnpackRequest} returns this
*/
proto.okapi.transport.v1.UnpackRequest.prototype.setSenderKey = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.UnpackRequest} returns this
 */
proto.okapi.transport.v1.UnpackRequest.prototype.clearSenderKey = function() {
  return this.setSenderKey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.UnpackRequest.prototype.hasSenderKey = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional okapi.keys.v1.JsonWebKey receiver_key = 2;
 * @return {?proto.okapi.keys.v1.JsonWebKey}
 */
proto.okapi.transport.v1.UnpackRequest.prototype.getReceiverKey = function() {
  return /** @type{?proto.okapi.keys.v1.JsonWebKey} */ (
    jspb.Message.getWrapperField(this, okapi_keys_v1_keys_pb.JsonWebKey, 2));
};


/**
 * @param {?proto.okapi.keys.v1.JsonWebKey|undefined} value
 * @return {!proto.okapi.transport.v1.UnpackRequest} returns this
*/
proto.okapi.transport.v1.UnpackRequest.prototype.setReceiverKey = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.UnpackRequest} returns this
 */
proto.okapi.transport.v1.UnpackRequest.prototype.clearReceiverKey = function() {
  return this.setReceiverKey(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.UnpackRequest.prototype.hasReceiverKey = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional pbmse.v1.EncryptedMessage message = 3;
 * @return {?proto.pbmse.v1.EncryptedMessage}
 */
proto.okapi.transport.v1.UnpackRequest.prototype.getMessage = function() {
  return /** @type{?proto.pbmse.v1.EncryptedMessage} */ (
    jspb.Message.getWrapperField(this, pbmse_v1_pbmse_pb.EncryptedMessage, 3));
};


/**
 * @param {?proto.pbmse.v1.EncryptedMessage|undefined} value
 * @return {!proto.okapi.transport.v1.UnpackRequest} returns this
*/
proto.okapi.transport.v1.UnpackRequest.prototype.setMessage = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.okapi.transport.v1.UnpackRequest} returns this
 */
proto.okapi.transport.v1.UnpackRequest.prototype.clearMessage = function() {
  return this.setMessage(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.okapi.transport.v1.UnpackRequest.prototype.hasMessage = function() {
  return jspb.Message.getField(this, 3) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.UnpackResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.UnpackResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.UnpackResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.UnpackResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    plaintext: msg.getPlaintext_asB64()
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.UnpackResponse}
 */
proto.okapi.transport.v1.UnpackResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.UnpackResponse;
  return proto.okapi.transport.v1.UnpackResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.UnpackResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.UnpackResponse}
 */
proto.okapi.transport.v1.UnpackResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setPlaintext(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.UnpackResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.UnpackResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.UnpackResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.UnpackResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getPlaintext_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      1,
      f
    );
  }
};


/**
 * optional bytes plaintext = 1;
 * @return {!(string|Uint8Array)}
 */
proto.okapi.transport.v1.UnpackResponse.prototype.getPlaintext = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * optional bytes plaintext = 1;
 * This is a type-conversion wrapper around `getPlaintext()`
 * @return {string}
 */
proto.okapi.transport.v1.UnpackResponse.prototype.getPlaintext_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getPlaintext()));
};


/**
 * optional bytes plaintext = 1;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getPlaintext()`
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.UnpackResponse.prototype.getPlaintext_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getPlaintext()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.okapi.transport.v1.UnpackResponse} returns this
 */
proto.okapi.transport.v1.UnpackResponse.prototype.setPlaintext = function(value) {
  return jspb.Message.setProto3BytesField(this, 1, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.okapi.transport.v1.CoreMessage.repeatedFields_ = [4];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.okapi.transport.v1.CoreMessage.prototype.toObject = function(opt_includeInstance) {
  return proto.okapi.transport.v1.CoreMessage.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.okapi.transport.v1.CoreMessage} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.CoreMessage.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    type: jspb.Message.getFieldWithDefault(msg, 2, ""),
    body: msg.getBody_asB64(),
    toList: (f = jspb.Message.getRepeatedField(msg, 4)) == null ? undefined : f,
    from: jspb.Message.getFieldWithDefault(msg, 5, ""),
    created: jspb.Message.getFieldWithDefault(msg, 6, 0),
    expires: jspb.Message.getFieldWithDefault(msg, 7, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.okapi.transport.v1.CoreMessage}
 */
proto.okapi.transport.v1.CoreMessage.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.okapi.transport.v1.CoreMessage;
  return proto.okapi.transport.v1.CoreMessage.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.okapi.transport.v1.CoreMessage} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.okapi.transport.v1.CoreMessage}
 */
proto.okapi.transport.v1.CoreMessage.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setType(value);
      break;
    case 3:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setBody(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.addTo(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setFrom(value);
      break;
    case 6:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setCreated(value);
      break;
    case 7:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setExpires(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.CoreMessage.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.okapi.transport.v1.CoreMessage.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.okapi.transport.v1.CoreMessage} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.okapi.transport.v1.CoreMessage.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getType();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getBody_asU8();
  if (f.length > 0) {
    writer.writeBytes(
      3,
      f
    );
  }
  f = message.getToList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      4,
      f
    );
  }
  f = message.getFrom();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = message.getCreated();
  if (f !== 0) {
    writer.writeInt64(
      6,
      f
    );
  }
  f = message.getExpires();
  if (f !== 0) {
    writer.writeInt64(
      7,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string type = 2;
 * @return {string}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getType = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.setType = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional bytes body = 3;
 * @return {!(string|Uint8Array)}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getBody = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * optional bytes body = 3;
 * This is a type-conversion wrapper around `getBody()`
 * @return {string}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getBody_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getBody()));
};


/**
 * optional bytes body = 3;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getBody()`
 * @return {!Uint8Array}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getBody_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getBody()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.setBody = function(value) {
  return jspb.Message.setProto3BytesField(this, 3, value);
};


/**
 * repeated string to = 4;
 * @return {!Array<string>}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getToList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 4));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.setToList = function(value) {
  return jspb.Message.setField(this, 4, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.addTo = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 4, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.clearToList = function() {
  return this.setToList([]);
};


/**
 * optional string from = 5;
 * @return {string}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getFrom = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.setFrom = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional int64 created = 6;
 * @return {number}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getCreated = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 6, 0));
};


/**
 * @param {number} value
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.setCreated = function(value) {
  return jspb.Message.setProto3IntField(this, 6, value);
};


/**
 * optional int64 expires = 7;
 * @return {number}
 */
proto.okapi.transport.v1.CoreMessage.prototype.getExpires = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {number} value
 * @return {!proto.okapi.transport.v1.CoreMessage} returns this
 */
proto.okapi.transport.v1.CoreMessage.prototype.setExpires = function(value) {
  return jspb.Message.setProto3IntField(this, 7, value);
};


goog.object.extend(exports, proto.okapi.transport.v1);
