// @trinsic/trinsic/lib/proto/services/verifiable-credentials/v1/verifiable-credentials_pb.js
// source: services/verifiable-credentials/v1/verifiable-credentials.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = Function('return this')();

var services_common_v1_common_pb = require('../../../services/common/v1/common_pb.js');
goog.object.extend(proto, services_common_v1_common_pb);
goog.exportSymbol('proto.services.verifiablecredentials.v1.CheckStatusRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.CheckStatusResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.CreateProofRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.CreateProofRequest.ProofCase', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.CreateProofResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.IssueFromTemplateRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.IssueFromTemplateResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.IssueRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.IssueResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.SendRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.SendRequest.DeliveryMethodCase', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.SendResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.UpdateStatusRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.UpdateStatusResponse', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.VerifyProofRequest', null, global);
goog.exportSymbol('proto.services.verifiablecredentials.v1.VerifyProofResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.IssueRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.IssueRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.IssueRequest.displayName = 'proto.services.verifiablecredentials.v1.IssueRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.IssueResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.IssueResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.IssueResponse.displayName = 'proto.services.verifiablecredentials.v1.IssueResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.IssueFromTemplateRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.displayName = 'proto.services.verifiablecredentials.v1.IssueFromTemplateRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.IssueFromTemplateResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.displayName = 'proto.services.verifiablecredentials.v1.IssueFromTemplateResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.CreateProofRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.services.verifiablecredentials.v1.CreateProofRequest.oneofGroups_);
};
goog.inherits(proto.services.verifiablecredentials.v1.CreateProofRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.CreateProofRequest.displayName = 'proto.services.verifiablecredentials.v1.CreateProofRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.CreateProofResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.CreateProofResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.CreateProofResponse.displayName = 'proto.services.verifiablecredentials.v1.CreateProofResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.VerifyProofRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.VerifyProofRequest.displayName = 'proto.services.verifiablecredentials.v1.VerifyProofRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.services.verifiablecredentials.v1.VerifyProofResponse.repeatedFields_, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.VerifyProofResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.VerifyProofResponse.displayName = 'proto.services.verifiablecredentials.v1.VerifyProofResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.SendRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_);
};
goog.inherits(proto.services.verifiablecredentials.v1.SendRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.SendRequest.displayName = 'proto.services.verifiablecredentials.v1.SendRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.SendResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.SendResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.SendResponse.displayName = 'proto.services.verifiablecredentials.v1.SendResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.UpdateStatusRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.UpdateStatusRequest.displayName = 'proto.services.verifiablecredentials.v1.UpdateStatusRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.UpdateStatusResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.UpdateStatusResponse.displayName = 'proto.services.verifiablecredentials.v1.UpdateStatusResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.CheckStatusRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.CheckStatusRequest.displayName = 'proto.services.verifiablecredentials.v1.CheckStatusRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.services.verifiablecredentials.v1.CheckStatusResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.services.verifiablecredentials.v1.CheckStatusResponse.displayName = 'proto.services.verifiablecredentials.v1.CheckStatusResponse';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.IssueRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.IssueRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.IssueRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.IssueRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    documentJson: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.IssueRequest}
 */
proto.services.verifiablecredentials.v1.IssueRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.IssueRequest;
  return proto.services.verifiablecredentials.v1.IssueRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.IssueRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.IssueRequest}
 */
proto.services.verifiablecredentials.v1.IssueRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setDocumentJson(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.IssueRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.IssueRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.IssueRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.IssueRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDocumentJson();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string document_json = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.IssueRequest.prototype.getDocumentJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.IssueRequest} returns this
 */
proto.services.verifiablecredentials.v1.IssueRequest.prototype.setDocumentJson = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.IssueResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.IssueResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.IssueResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.IssueResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    signedDocumentJson: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.IssueResponse}
 */
proto.services.verifiablecredentials.v1.IssueResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.IssueResponse;
  return proto.services.verifiablecredentials.v1.IssueResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.IssueResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.IssueResponse}
 */
proto.services.verifiablecredentials.v1.IssueResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setSignedDocumentJson(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.IssueResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.IssueResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.IssueResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.IssueResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSignedDocumentJson();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string signed_document_json = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.IssueResponse.prototype.getSignedDocumentJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.IssueResponse} returns this
 */
proto.services.verifiablecredentials.v1.IssueResponse.prototype.setSignedDocumentJson = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.IssueFromTemplateRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    templateId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    valuesJson: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.IssueFromTemplateRequest}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.IssueFromTemplateRequest;
  return proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.IssueFromTemplateRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.IssueFromTemplateRequest}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setTemplateId(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setValuesJson(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.IssueFromTemplateRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTemplateId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getValuesJson();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string template_id = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.prototype.getTemplateId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.IssueFromTemplateRequest} returns this
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.prototype.setTemplateId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string values_json = 2;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.prototype.getValuesJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.IssueFromTemplateRequest} returns this
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateRequest.prototype.setValuesJson = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.IssueFromTemplateResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    documentJson: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.IssueFromTemplateResponse}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.IssueFromTemplateResponse;
  return proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.IssueFromTemplateResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.IssueFromTemplateResponse}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setDocumentJson(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.IssueFromTemplateResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDocumentJson();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string document_json = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.prototype.getDocumentJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.IssueFromTemplateResponse} returns this
 */
proto.services.verifiablecredentials.v1.IssueFromTemplateResponse.prototype.setDocumentJson = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.oneofGroups_ = [[2,3]];

/**
 * @enum {number}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.ProofCase = {
  PROOF_NOT_SET: 0,
  ITEM_ID: 2,
  DOCUMENT_JSON: 3
};

/**
 * @return {proto.services.verifiablecredentials.v1.CreateProofRequest.ProofCase}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.getProofCase = function() {
  return /** @type {proto.services.verifiablecredentials.v1.CreateProofRequest.ProofCase} */(jspb.Message.computeOneofCase(this, proto.services.verifiablecredentials.v1.CreateProofRequest.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.CreateProofRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.CreateProofRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    revealDocumentJson: jspb.Message.getFieldWithDefault(msg, 1, ""),
    itemId: jspb.Message.getFieldWithDefault(msg, 2, ""),
    documentJson: jspb.Message.getFieldWithDefault(msg, 3, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.CreateProofRequest}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.CreateProofRequest;
  return proto.services.verifiablecredentials.v1.CreateProofRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.CreateProofRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.CreateProofRequest}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setRevealDocumentJson(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setItemId(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setDocumentJson(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.CreateProofRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.CreateProofRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getRevealDocumentJson();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeString(
      2,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 3));
  if (f != null) {
    writer.writeString(
      3,
      f
    );
  }
};


/**
 * optional string reveal_document_json = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.getRevealDocumentJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.CreateProofRequest} returns this
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.setRevealDocumentJson = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string item_id = 2;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.getItemId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.CreateProofRequest} returns this
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.setItemId = function(value) {
  return jspb.Message.setOneofField(this, 2, proto.services.verifiablecredentials.v1.CreateProofRequest.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.services.verifiablecredentials.v1.CreateProofRequest} returns this
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.clearItemId = function() {
  return jspb.Message.setOneofField(this, 2, proto.services.verifiablecredentials.v1.CreateProofRequest.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.hasItemId = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional string document_json = 3;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.getDocumentJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.CreateProofRequest} returns this
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.setDocumentJson = function(value) {
  return jspb.Message.setOneofField(this, 3, proto.services.verifiablecredentials.v1.CreateProofRequest.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.services.verifiablecredentials.v1.CreateProofRequest} returns this
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.clearDocumentJson = function() {
  return jspb.Message.setOneofField(this, 3, proto.services.verifiablecredentials.v1.CreateProofRequest.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.services.verifiablecredentials.v1.CreateProofRequest.prototype.hasDocumentJson = function() {
  return jspb.Message.getField(this, 3) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.CreateProofResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.CreateProofResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.CreateProofResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.CreateProofResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    proofDocumentJson: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.CreateProofResponse}
 */
proto.services.verifiablecredentials.v1.CreateProofResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.CreateProofResponse;
  return proto.services.verifiablecredentials.v1.CreateProofResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.CreateProofResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.CreateProofResponse}
 */
proto.services.verifiablecredentials.v1.CreateProofResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setProofDocumentJson(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.CreateProofResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.CreateProofResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.CreateProofResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.CreateProofResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProofDocumentJson();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string proof_document_json = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.CreateProofResponse.prototype.getProofDocumentJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.CreateProofResponse} returns this
 */
proto.services.verifiablecredentials.v1.CreateProofResponse.prototype.setProofDocumentJson = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.VerifyProofRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.VerifyProofRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    proofDocumentJson: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofRequest}
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.VerifyProofRequest;
  return proto.services.verifiablecredentials.v1.VerifyProofRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.VerifyProofRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofRequest}
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setProofDocumentJson(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.VerifyProofRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.VerifyProofRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getProofDocumentJson();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string proof_document_json = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest.prototype.getProofDocumentJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofRequest} returns this
 */
proto.services.verifiablecredentials.v1.VerifyProofRequest.prototype.setProofDocumentJson = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.VerifyProofResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.VerifyProofResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    isValid: jspb.Message.getBooleanFieldWithDefault(msg, 1, false),
    validationMessagesList: (f = jspb.Message.getRepeatedField(msg, 2)) == null ? undefined : f
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofResponse}
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.VerifyProofResponse;
  return proto.services.verifiablecredentials.v1.VerifyProofResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.VerifyProofResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofResponse}
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setIsValid(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.addValidationMessages(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.VerifyProofResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.VerifyProofResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getIsValid();
  if (f) {
    writer.writeBool(
      1,
      f
    );
  }
  f = message.getValidationMessagesList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      2,
      f
    );
  }
};


/**
 * optional bool is_valid = 1;
 * @return {boolean}
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.prototype.getIsValid = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 1, false));
};


/**
 * @param {boolean} value
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofResponse} returns this
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.prototype.setIsValid = function(value) {
  return jspb.Message.setProto3BooleanField(this, 1, value);
};


/**
 * repeated string validation_messages = 2;
 * @return {!Array<string>}
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.prototype.getValidationMessagesList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 2));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofResponse} returns this
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.prototype.setValidationMessagesList = function(value) {
  return jspb.Message.setField(this, 2, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofResponse} returns this
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.prototype.addValidationMessages = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 2, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.services.verifiablecredentials.v1.VerifyProofResponse} returns this
 */
proto.services.verifiablecredentials.v1.VerifyProofResponse.prototype.clearValidationMessagesList = function() {
  return this.setValidationMessagesList([]);
};



/**
 * Oneof group definitions for this message. Each group defines the field
 * numbers belonging to that group. When of these fields' value is set, all
 * other fields in the group are cleared. During deserialization, if multiple
 * fields are encountered for a group, only the last value seen will be kept.
 * @private {!Array<!Array<number>>}
 * @const
 */
proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_ = [[1,2,3]];

/**
 * @enum {number}
 */
proto.services.verifiablecredentials.v1.SendRequest.DeliveryMethodCase = {
  DELIVERY_METHOD_NOT_SET: 0,
  EMAIL: 1,
  DID_URI: 2,
  DIDCOMM_INVITATION_JSON: 3
};

/**
 * @return {proto.services.verifiablecredentials.v1.SendRequest.DeliveryMethodCase}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.getDeliveryMethodCase = function() {
  return /** @type {proto.services.verifiablecredentials.v1.SendRequest.DeliveryMethodCase} */(jspb.Message.computeOneofCase(this, proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_[0]));
};



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.SendRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.SendRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.SendRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    email: jspb.Message.getFieldWithDefault(msg, 1, ""),
    didUri: jspb.Message.getFieldWithDefault(msg, 2, ""),
    didcommInvitationJson: jspb.Message.getFieldWithDefault(msg, 3, ""),
    documentJson: jspb.Message.getFieldWithDefault(msg, 100, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.SendRequest}
 */
proto.services.verifiablecredentials.v1.SendRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.SendRequest;
  return proto.services.verifiablecredentials.v1.SendRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.SendRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.SendRequest}
 */
proto.services.verifiablecredentials.v1.SendRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setEmail(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setDidUri(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setDidcommInvitationJson(value);
      break;
    case 100:
      var value = /** @type {string} */ (reader.readString());
      msg.setDocumentJson(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.SendRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.SendRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.SendRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = /** @type {string} */ (jspb.Message.getField(message, 1));
  if (f != null) {
    writer.writeString(
      1,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeString(
      2,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 3));
  if (f != null) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getDocumentJson();
  if (f.length > 0) {
    writer.writeString(
      100,
      f
    );
  }
};


/**
 * optional string email = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.getEmail = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.SendRequest} returns this
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.setEmail = function(value) {
  return jspb.Message.setOneofField(this, 1, proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.services.verifiablecredentials.v1.SendRequest} returns this
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.clearEmail = function() {
  return jspb.Message.setOneofField(this, 1, proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.hasEmail = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional string did_uri = 2;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.getDidUri = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.SendRequest} returns this
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.setDidUri = function(value) {
  return jspb.Message.setOneofField(this, 2, proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.services.verifiablecredentials.v1.SendRequest} returns this
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.clearDidUri = function() {
  return jspb.Message.setOneofField(this, 2, proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.hasDidUri = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional string didcomm_invitation_json = 3;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.getDidcommInvitationJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.SendRequest} returns this
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.setDidcommInvitationJson = function(value) {
  return jspb.Message.setOneofField(this, 3, proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_[0], value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.services.verifiablecredentials.v1.SendRequest} returns this
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.clearDidcommInvitationJson = function() {
  return jspb.Message.setOneofField(this, 3, proto.services.verifiablecredentials.v1.SendRequest.oneofGroups_[0], undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.hasDidcommInvitationJson = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional string document_json = 100;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.getDocumentJson = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 100, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.SendRequest} returns this
 */
proto.services.verifiablecredentials.v1.SendRequest.prototype.setDocumentJson = function(value) {
  return jspb.Message.setProto3StringField(this, 100, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.SendResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.SendResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.SendResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.SendResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    status: jspb.Message.getFieldWithDefault(msg, 1, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.SendResponse}
 */
proto.services.verifiablecredentials.v1.SendResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.SendResponse;
  return proto.services.verifiablecredentials.v1.SendResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.SendResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.SendResponse}
 */
proto.services.verifiablecredentials.v1.SendResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.services.common.v1.ResponseStatus} */ (reader.readEnum());
      msg.setStatus(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.SendResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.SendResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.SendResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.SendResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStatus();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
};


/**
 * optional services.common.v1.ResponseStatus status = 1;
 * @return {!proto.services.common.v1.ResponseStatus}
 */
proto.services.verifiablecredentials.v1.SendResponse.prototype.getStatus = function() {
  return /** @type {!proto.services.common.v1.ResponseStatus} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.services.common.v1.ResponseStatus} value
 * @return {!proto.services.verifiablecredentials.v1.SendResponse} returns this
 */
proto.services.verifiablecredentials.v1.SendResponse.prototype.setStatus = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.UpdateStatusRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.UpdateStatusRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    credentialStatusId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    revoked: jspb.Message.getBooleanFieldWithDefault(msg, 2, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.UpdateStatusRequest}
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.UpdateStatusRequest;
  return proto.services.verifiablecredentials.v1.UpdateStatusRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.UpdateStatusRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.UpdateStatusRequest}
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setCredentialStatusId(value);
      break;
    case 2:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setRevoked(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.UpdateStatusRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.UpdateStatusRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCredentialStatusId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getRevoked();
  if (f) {
    writer.writeBool(
      2,
      f
    );
  }
};


/**
 * optional string credential_status_id = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.prototype.getCredentialStatusId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.UpdateStatusRequest} returns this
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.prototype.setCredentialStatusId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional bool revoked = 2;
 * @return {boolean}
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.prototype.getRevoked = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 2, false));
};


/**
 * @param {boolean} value
 * @return {!proto.services.verifiablecredentials.v1.UpdateStatusRequest} returns this
 */
proto.services.verifiablecredentials.v1.UpdateStatusRequest.prototype.setRevoked = function(value) {
  return jspb.Message.setProto3BooleanField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.UpdateStatusResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.UpdateStatusResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    status: jspb.Message.getFieldWithDefault(msg, 1, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.UpdateStatusResponse}
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.UpdateStatusResponse;
  return proto.services.verifiablecredentials.v1.UpdateStatusResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.UpdateStatusResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.UpdateStatusResponse}
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.services.common.v1.ResponseStatus} */ (reader.readEnum());
      msg.setStatus(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.UpdateStatusResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.UpdateStatusResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStatus();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
};


/**
 * optional services.common.v1.ResponseStatus status = 1;
 * @return {!proto.services.common.v1.ResponseStatus}
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse.prototype.getStatus = function() {
  return /** @type {!proto.services.common.v1.ResponseStatus} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.services.common.v1.ResponseStatus} value
 * @return {!proto.services.verifiablecredentials.v1.UpdateStatusResponse} returns this
 */
proto.services.verifiablecredentials.v1.UpdateStatusResponse.prototype.setStatus = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.CheckStatusRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.CheckStatusRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    credentialStatusId: jspb.Message.getFieldWithDefault(msg, 1, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.CheckStatusRequest}
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.CheckStatusRequest;
  return proto.services.verifiablecredentials.v1.CheckStatusRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.CheckStatusRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.CheckStatusRequest}
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setCredentialStatusId(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.CheckStatusRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.CheckStatusRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCredentialStatusId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
};


/**
 * optional string credential_status_id = 1;
 * @return {string}
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest.prototype.getCredentialStatusId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.services.verifiablecredentials.v1.CheckStatusRequest} returns this
 */
proto.services.verifiablecredentials.v1.CheckStatusRequest.prototype.setCredentialStatusId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.services.verifiablecredentials.v1.CheckStatusResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.services.verifiablecredentials.v1.CheckStatusResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    revoked: jspb.Message.getBooleanFieldWithDefault(msg, 1, false)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.services.verifiablecredentials.v1.CheckStatusResponse}
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.services.verifiablecredentials.v1.CheckStatusResponse;
  return proto.services.verifiablecredentials.v1.CheckStatusResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.services.verifiablecredentials.v1.CheckStatusResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.services.verifiablecredentials.v1.CheckStatusResponse}
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setRevoked(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.services.verifiablecredentials.v1.CheckStatusResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.services.verifiablecredentials.v1.CheckStatusResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getRevoked();
  if (f) {
    writer.writeBool(
      1,
      f
    );
  }
};


/**
 * optional bool revoked = 1;
 * @return {boolean}
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse.prototype.getRevoked = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 1, false));
};


/**
 * @param {boolean} value
 * @return {!proto.services.verifiablecredentials.v1.CheckStatusResponse} returns this
 */
proto.services.verifiablecredentials.v1.CheckStatusResponse.prototype.setRevoked = function(value) {
  return jspb.Message.setProto3BooleanField(this, 1, value);
};


goog.object.extend(exports, proto.services.verifiablecredentials.v1);
